flink/flink-scala-parent/src/main/scala/org/apache/zeppelin/flink/internal/FlinkILoop.scala

```diff
@@ -101,9 +104,12 @@ class FlinkILoop(
   private val tmpDirBase: File = {
     // get unique temporary folder:
     val abstractID: String = new AbstractID().toString
-    val tmpDir: File = new File(
-      System.getProperty("java.io.tmpdir"),
-      "scala_shell_tmp-" + abstractID)
+    var scalaShellTmpParentFolder = flinkScalaInterpreter.properties.getProperty("zeppelin.flink.scala.shell.tmp_dir")
+    if (StringUtils.isBlank(scalaShellTmpParentFolder)) {
+      scalaShellTmpParentFolder = System.getProperty("java.io.tmpdir")
+    }
+    val tmpDir: File = new File(scalaShellTmpParentFolder, "scala_shell_tmp-" + abstractID)
+    LOGGER.info("Folder for scala shell compiled jar: {}", tmpDir.getAbsolutePath)
     if (!tmpDir.exists) {
       tmpDir.mkdir
```

Comment:
Is `mkdirs` more appropriate than `mkdir`

Reply:
Make sense, fixed